torch/_inductor/kernel/mm_scaled_grouped.py

```diff
@@ -0,0 +1,292 @@
+            while tidx >= iterated_tiles and tidx < iterated_tiles + num_tiles:
+                gidx = tidx - iterated_tiles
+                # Split M first and N second.
+                tile_m_idx = gidx % num_m_tiles
+                tile_n_idx = gidx // num_m_tiles
+
+                accumulator = tl.zeros((BLOCK_M, BLOCK_N), dtype=tl.float32)
+                tl.static_assert(K % BLOCK_K == 0)
+                if USE_TMA_LOAD:
+                    m_offset = (M_start_offset + tile_m_idx * BLOCK_M).to(tl.int32)
+                    n_offset = (N_start_offset + tile_n_idx * BLOCK_N).to(tl.int32)
+                    for k_offset in range(0, K, BLOCK_K):
+                        a = tl._experimental_descriptor_load(
+                            a_desc_ptr,
+                            [m_offset, k_offset],
+                            [BLOCK_M, BLOCK_K],
+                            dtype,
+                        )
+                        b = tl._experimental_descriptor_load(
+                            b_desc_ptr,
+                            [n_offset, k_offset],
+                            [BLOCK_N, BLOCK_K],
+                            dtype,
+                        )
+                        if USE_FAST_ACCUM:
+                            accumulator = tl.dot(a, b.T, accumulator)
+                        else:
+                            accumulator += tl.dot(a, b.T)
+                else:
+                    offs_am = tile_m_idx * BLOCK_M + tl.arange(0, BLOCK_M)
+                    offs_bn = tile_n_idx * BLOCK_N + tl.arange(0, BLOCK_N)
+                    offs_k = tl.arange(0, BLOCK_K)
+                    a_ptrs = (
+                        a_desc_ptr
+                        + (M_start_offset + offs_am[:, None]) * K
+                        + offs_k[None, :]
+                    )
+                    b_ptrs = (
+                        b_desc_ptr
+                        + (N_start_offset + offs_bn[:, None]) * K
+                        + offs_k[None, :]
+                    )
+                    for k_offset in range(0, K, BLOCK_K):
+                        a = tl.load(a_ptrs, mask=offs_am[:, None] < m_size)
+                        b = tl.load(b_ptrs, mask=offs_bn[:, None] < n_size)
+                        accumulator += tl.dot(a, b.T)
+                        a_ptrs += BLOCK_K
+                        b_ptrs += BLOCK_K
+
+                offs_am = tile_m_idx * BLOCK_M + tl.arange(0, BLOCK_M)
+                offs_bn = tile_n_idx * BLOCK_N + tl.arange(0, BLOCK_N)
+                a_scale = tl.load(
+                    a_scale_ptr + M_start_offset + offs_am[:, None],
+                    mask=offs_am[:, None] < m_size,
+                )
+                b_scale = tl.load(
+                    b_scale_ptr + N_start_offset + offs_bn[None, :],
+                    mask=offs_bn[None, :] < n_size,
+                )
+                c = accumulator.to(tl.float32) * a_scale * b_scale
+
+                if USE_TMA_STORE:
```

Comment:
I don't like this USE_TMA_STORE option (i know we are not using it) because it can't do masking, and we absolutely need m masking (that regular store is doing below) because we might have computed completely bogus results with m_size that's not a multiple of required tma size.

Reply:
Interesting.   This is actually a little surprising to me, because I thought the way the TMA store descriptor was being set up accounts for the m_size... but I'm also not sure how that interacts with the required alignment/size.

Since inductor currently can't use this anyways though, I'm thinking I'll just delete the  `USE_TMA_STORE` branch, since there's no reason to leave around dead code